src/main/java/com/commercetools/sync/services/impl/UnresolvedReferencesServiceImpl.java

```diff
@@ -37,6 +39,7 @@ public UnresolvedReferencesServiceImpl(@Nonnull final ProductSyncOptions baseSyn
     @Nonnull
     @Override
     public CompletionStage<Set<WaitingToBeResolved>> fetch(@Nonnull final Set<String> keys) {
+        keys.remove(EMPTY);
```

Comment:
- It's not a good practice to mutate in the user input.
- What is the purpose of removing `EMPTY` at all? We should not be expecting it to be passed.

Reply:
3dea2d39e4f5b8ddd76371f5a685e88647732b0a